### PR TITLE
collections homework

### PR DESCRIPTION
**Pull request by IvanGyulumyan99**

### CODE REVIEW

File: src/collections/main/Main.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        arrayList.add(arthur);
+        arrayList.add(narek22);
+        arrayList.add(ivan18);
+        StudentNameAscendingAndAgeDescendingComparator cmp = new StudentNameAscendingAndAgeDescendingComparator();
+        arrayList.sort(cmp);
+        for (Student tmp : arrayList) {
+            System.out.println(tmp);
+        }
+
+        dynamicArrayOfStudents.get(-11);
```
Comment (vahramepam) [inline src/collections/main/Main.java]: I tried to remove `anna = new Student("Anna", "Martirosyan", 18)` from `dynamicArrayOfStudents` and it returned false, why?
`dynamicArrayOfStudents.remove(new Student("Anna", "Martirosyan", 18))`

File: src/collections/comparator/StudentNameAscendingAndAgeDescendingComparator.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package collections.comparator;
+
+import collections.model.Student;
+import java.util.Comparator;
+
+public class StudentNameAscendingAndAgeDescendingComparator implements Comparator<Student> {
+    @Override
+    public int compare(Student student1, Student student2) {
+        int nameCompare = student1.getFirstName().compareTo(student2.getFirstName());
+        int ageCompare = student2.getAge() - student1.getAge();
```
Comment (vahramepam) [inline src/collections/comparator/StudentNameAscendingAndAgeDescendingComparator.java]: `ageCompare` is always calculated, but only needed when `nameCompare == 0`

File: src/collections/main/Main.java (hunk 1 above)
Comment (IvanGyulumyan99) [reply src/collections/main/Main.java]: it returns false only when the object we want to remove doesn't exists in dynamic array, in this case we want to remove new object that doesn't exists in our dynamic array, so it returns false

Comment (vahramepam) [reply src/collections/main/Main.java]: What do you mean by `new object`? Aren't these two objects the same? 
added: `new Student("Anna", "Martirosyan", 18)`
removed: `new Student("Anna", "Martirosyan", 18)`

Comment (IvanGyulumyan99) [reply src/collections/main/Main.java]: this objects have different references, so they are different persons, I haven't overrided equals method for Students.

Comment (vahramepam) [reply src/collections/main/Main.java]: Right, and that is the exact point I am trying to make clear for you. Unless you properly override the equals method, your remove method will not work properly, e.g. suppose you created a `dynamicArrayOfStudents` somewhere in your project, and tried to use it in any other place, you have to send them not only your `dynamicArrayOfStudents` but also all references you created, in order your remove method work properly.

**Pull request merged**